core/templates/dev/head/components/solution_editor_directive.html

```diff
@@ -0,0 +1,103 @@
+<div style="height: 100%;">
+  <div ng-if="!solutionEditorIsOpen"
+       style="height: 100%;"
+       ng-attr-title="<[isEditable ? 'Edit solution' : '']>">
+    <div class="oppia-readonly-rule-tile"
+         ng-class="{'oppia-editable-section': isEditable}">
+      <div class="oppia-rule-preview oppia-transition-200">
+        <div class="oppia-click-to-start-editing"
+             ng-if="isEditable" ng-click="openSolutionEditor()">
+          <i ng-if="isEditable"
+             class="material-icons oppia-editor-edit-icon pull-right"
+             title="Edit Solution">&#xE254;
+          </i>
+        </div>
+        <strong>
+          <[solution.answerIsExclusive ? 'Only' : 'One']> Solution is...
+        </strong>
+        <span ng-if="objectType !== 'CodeString' &&
```

Comment:
There's a lot of stuff going on here. Can we instead define a helper function in the JS that takes in the object type and outputs the thing to display? Perhaps on the Solution object itself? That way, you can test it too.